src/ChangeStream.php

```diff
@@ -17,38 +17,28 @@
 
 namespace MongoDB;
 
-use MongoDB\ChangeStreamIterator;
-use MongoDB\Driver\Manager;
-use MongoDB\Exception\ResumableException;
-use MongoDB\Operation\ChangeStreamCommand;
+use MongoDB\Driver\Cursor;
+use MongoDB\Driver\Exception\RuntimeException;
+use IteratorIterator;
+use Iterator;
 
 /**
- * Operation for the changeStream command.
+ * Iterator for the changeStream command.
  *
  * @api
  * @see \MongoDB\Collection::changeStream()
  * @see http://docs.mongodb.org/manual/reference/command/changeStream/
  */
-class ChangeStream
+class ChangeStream implements Iterator
 {
-    private $databaseName;
-    private $collectionName;
-    private $pipeline;
-    private $options;
     private $resumeToken;
-    private $manager;
+    private $resumeCallable;
     private $csIt;
 
-    public function __construct($cursor, $databaseName, $collectionName, array $pipeline, array $options = [], $resumeToken, Manager $manager)
+    public function __construct(Cursor $cursor, Callable $resumeCallable)
```

Comment:
I know class names in PHP are case-insensitive, but I'm not sure about other type names. [`callable`](http://php.net/manual/en/language.types.callable.php) is the canonical casing, though.